File: LAB3/series_sum.cpp
/**
 * @file series_sum.cpp
 * @brief Вычисление суммы бесконечного ряда вида ∑(n^a / b^n).
 * 
 * Программа принимает два целых числа a и b (1 <= a, b <= 10).
 * Если ряд расходится (b = 1), выводится "infinity".
 * Если сумма рациональна, выводится несократимая дробь.
 * Для a > 2 вывод "irrational" (требуется расширение формул).
 * 
 * Примеры:
 * Ввод: 1 2 → Вывод: 2/1
 * Ввод: 2 1 → Вывод: infinity
 */

 #include <iostream>
 #include <algorithm> 
 
 using namespace std;
 
 /**
  * @brief Вычисляет наибольший общий делитель (НОД) двух целых чисел.
  * @param a Первое число.
  * @param b Второе число.
  * @return НОД(a, b).
  */
 int gcd(int a, int b) {
     while (b != 0) {
         int temp = b;
         b = a % b;
         a = temp;
     }
     return a;
 }
 
 /**
  * @brief Основная функция программы.
  * 
  * Считывает входные данные, проверяет условия сходимости, 
  * вычисляет сумму ряда и выводит результат.
  */
 int main() {
     int a, b; ///< Входные числа: a и b (1 <= a, b <= 10)
     cin >> a >> b;
 
     if (b == 1) {
         cout << "infinity" << endl;
         return 0;
     }
 
     int numerator;   ///< Числитель дроби суммы ряда
     int denominator; ///< Знаменатель дроби суммы ряда
 
     // Вычисление числителя и знаменателя в зависимости от a
     switch (a) {
         case 1:
             numerator = b;
             denominator = (b - 1) * (b - 1);
             break;
         case 2:
             numerator = b * (b + 1);
             denominator = (b - 1) * (b - 1) * (b - 1);
             break;
         default:
             // Для a > 2 требуется реализация дополнительных формул
             cout << "irrational" << endl;
             return 0;
     }
 
     // Сокращение дроби
     int common_divisor = gcd(numerator, denominator);
     numerator /= common_divisor;
     denominator /= common_divisor;
 
     cout << numerator << "/" << denominator << endl;
 
     return 0;
 }
 

File: LAB3/series_sum.py
import math

def gcd(a: int, b: int) -> int:
    """
    Вычисляет наибольший общий делитель (НОД) двух целых чисел.

    Параметры:
        a (int): Первое число.
        b (int): Второе число.

    Возвращает:
        int: Наибольший общий делитель чисел a и b.
    """
    while b:
        a, b = b, a % b
    return a

# Считывание входных данных
a, b = map(int, input().split())

if b == 1:
    print("infinity")
else:
    """
    Вычисляет сумму ряда ∑(n^a / b^n) для заданных a и b.
    
    Условия:
        - 1 <= a, b <= 10
        - При b = 1 ряд расходится (вывод 'infinity').
        - Для a > 2 требуется расширение формул (временная заглушка: 1/1).

    Формулы:
        - a = 1: S = b / (b-1)^2
        - a = 2: S = b(b+1) / (b-1)^3
    """
    if a == 1:
        numerator = b
        denominator = (b - 1) ** 2
    elif a == 2:
        numerator = b * (b + 1)
        denominator = (b - 1) ** 3
    else:
        # Заглушка для a > 2 (требуется реализация)
        numerator = 1
        denominator = 1

    # Сокращение дроби
    common_divisor = gcd(numerator, denominator)
    numerator //= common_divisor
    denominator //= common_divisor

    print(f"{numerator}/{denominator}")
    